mavros/src/lib/uas_data.cpp

```diff
@@ -40,13 +47,15 @@ UAS::UAS() :
 	fcu_capabilities(0)
 {
 	try {
-		// Using smakkest dataset with 5' grid,
+		// Using smallest dataset with 5' grid,
 		// From default location,
 		// Use cubic interpolation, Thread safe
 		egm96_5 = std::make_shared<GeographicLib::Geoid>("egm96-5", "", true, true);
 	}
 	catch (const std::exception &e) {
+		// catch exception and shutdown node
 		ROS_ERROR_STREAM("UAS: GeographicLib exception: " << e.what());
+		signal(SIGINT, term);
 	}
```

Comment:
But that it not what you wrote in code.
Code:
1. try to load egm96-5
2. on error:
    1. Log error
    2. Set SIGINT signal handler. `man signal`